frontend/src/app/forms/steps/basic-info/basic-info.component.ts

```diff
@@ -0,0 +1,71 @@
+/*
+@Author
+Bharathwaj Ravi
+
+Add modifiers under @Modifiers
+@Modifiers
+
+@Purpose
+  - This component renders the basic information step view and it's actions.
+*/
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+import { FormGroup } from './../../../common/forms/forms';
+import { BasicInfo } from './../../../interfaces/basic-info';
+
+@Component({
+  selector: 'form-basic-info',
+  templateUrl: './basic-info.component.html',
+  styleUrls: ['./basic-info.component.scss'],
+})
+export class BasicInfoComponent implements OnInit {
+  /*
+  @Usage It holds the current menu object from parent
+  */
+  @Input() menuObject: {
+    id: number;
+    displayName: string;
+    isCompleted: boolean;
+    isActive: boolean;
+    selecteor: string;
+  };
+
+  /*
+  @Usage It holds typesafe form group property fields.
+  */
+  @Input() basicInfoObject: FormGroup<BasicInfo>;
+
+  /*
+  @Usage It emits the progress callback on this step.
+  */
+  @Output() pageProgressStatusCallBack = new EventEmitter();
+
+  /*
+  @Usage It holds the array objects of drop down menu.
+  */
+  salutationsArray = [
+    { value: 'mr', displayName: 'Mr' },
+    { value: 'mrs', displayName: 'Mrs' },
+    { value: 'ms', displayName: 'Ms' },
+  ];
+
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  /*
+  @Usage It handles the on change event for the fields and send emit call back event to parent.
+  */
+  onValChange(): void {
+    if (
+      this.basicInfoObject.controls.salutation.value.length &&
+      this.basicInfoObject.controls.firstName.value.length &&
+      this.basicInfoObject.controls.lastName.value.length
```

Comment:
In the future, we might think about using Validators. But let's ignore that for now. (Validators basically allow you to, for instance, mark fields as required. And then, you simply ask your form element "Are you valid?" or "Are all of your child controls valid?"

Reply:
Ya we have to think of that. Also will be better if there is a getter setter method that can get and set value. It is pretty hard to access with object.controls.

Reply:
Usually, you don't need to manually set the values. But if you need to (maybe because you stored some stuff in localstorge), then setValue or patchValue on the FormGroup is your friend :)